pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilterWithClassLoader.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service.plugin;
+
+import org.apache.bookkeeper.mledger.Entry;
+import org.apache.pulsar.common.nar.NarClassLoader;
+
+public class EntryFilterWithClassLoader implements EntryFilter {
+    private final EntryFilter entryFilter;
+    private final NarClassLoader classLoader;
+
+    public EntryFilterWithClassLoader(EntryFilter entryFilter, NarClassLoader classLoader) {
+        this.entryFilter = entryFilter;
+        this.classLoader = classLoader;
+    }
+
+    @Override
+    public FilterResult filterEntry(Entry entry, FilterContext context) {
+        return entryFilter.filterEntry(entry, context);
+    }
```

Comment:
Please add a close() method that closed the Classloader and then calls the close method of the wrapper object.
Otherwise we have a leak

Reply:
Good catch